### DISCUSSION TITLE
Using "parameter: List[str] = Form(...)" produces unusable Swagger

### DISCUSSION THREAD

**Discussion by othi** [fastapi/fastapi#9090] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI, Form, Response
from typing import List

app = FastAPI()

@app.put("/test")
async def test(mylist: List[str] = Form(...)):
  return Response(content=str(mylist))
```
Don't forget to install python-multipart to support Form parsing.

### Description

* Open the browser at `/docs`,  add two entries `a` and `b` in the Swagger UI and call the endpoint `/test`.
* It returns the string representation of the parameter mylist: `['a,b']`
* But I expected it to return `['a', 'b']`.

The Swagger UI generates the API call like this (I put the relevant -d parameter first for legibility):
```bash
curl -d "mylist=a,b" -X PUT "http://127.0.0.1:8081/test" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" 
```
It should generate it like this in order to be parsed correctly by FastAPI:
```bash
curl -d "mylist=a&mylist=b" -X PUT "http://127.0.0.1:8081/test" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" 
```

### Environment

* OS: Linux
* FastAPI Version: 0.58.1

* Python version: 3.6.9

### Additional context

This does work correctly for Query parameters. After replacing `Form`  with `Query` in the example above, Swagger generates the following API call:
```bash
curl -X PUT "http://127.0.0.1:8081/test?mylist=a&mylist=b" -H "accept: application/json"
```
in which case the route correctly outputs `['a', 'b']`.

This *probably* has something to do with [the OpenAPI serialization option](https://swagger.io/docs/specification/serialization/), but I am not versed enough on this topic to be sure.

**Comment by samjoy** [comment]
Any idea how to solve this issue? I am facing the same thing.

**Comment by sangm** [comment]
Same issue here!

I don't see an option to provide `explode=True` to the form param!

**Comment by batmanscode** [comment]
Having the same problem! Was having a hard time trying to figure out what I did wrong :( I eventually traced it back to this

Couldn't find much on this issue. Perhaps it's not a common way `Form()` is used

**Comment by JarroVGIT** [comment]
This is not a FastAPI issue but a SwaggerUI issue, no?

have you tried in Redoc?

**Comment by iudeen** [comment]
> have you tried in Redoc?

Redoc does not have "Try it" feature. 

> The Try it feature is not available in Redoc. You will need access to Redocly API docs or Developer portal products to use the Try it feature.

And yes, its a SwaggerUI issue IMO.

**Comment by arLevi** [comment]
I was able to solve my issue by remove the `Form()`
for some reason, I thought that it was mandatory to tell FastAPI to grab the right field from the Form
But it seems to work w/o it as well (?!)

Instead of:
```
async def test(request: Request,
               fieldA : str = Form('fieldA'),
               mylist : list = Form('mylist')):

# This gives the error in the console:
SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
```

I wrote this:
```
async def test(request: Request,
               fieldA : str = Form('fieldA'),
               mylist : list = []):
```

SwaggerUI that is being rendered by FastAPI - opened the endpoint correctly.
And it grabbed the right field from the Form field correctly...

So I wonder - why I need to define these `Form('fieldX')` everywhere ? ....
And this is why I believe this is a FastAPI issue vs a Swagger UI issue ...
